### PR TITLE
Fixed some compiler warnings in applet.

### PR DESCRIPTION
**Pull request by KG5IEN**
See issue #704 

I didn't tackle the #warnings in the source, but everything else is now compiler warning free.

Testing done:
made ci, scanned output for compiler warnings that weren't from #warnings.

### CODE REVIEW

**Comment by KG5IEN** [comment]
Note: Updated request resolves merge conflicts but does not address new warnings introduced to applet in 8f8ca60.

File: applet/src/gfx.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ void green_led(int on);
 void red_led(int on);
 void lcd_background_led(int on);
 
-void print_time_hook(void);
+void print_time_hook(char *log);
```
Comment (KG5IEN) [inline applet/src/gfx.h]: This particular fix matches the existing actual parameter list of print_time_hook from gfx.c.

If issue #732 is correct, then both gfx.c and gfx.h will need to change in this PR.

**Comment by Wolf-DL4YHF** [comment]
Hmm... but as I wrote (assumed), the intention of the parameter "log" is 
*NOT* being a char pointer, but a single char.
So this fix would be wrong. Again, I *guess* what Mike meant was a 
single char, i.e.:

void print_time_hook( char log ).
Note the lack of the asterisk.

Cheers,
   Wolf .

Am 18.04.2017 um 04:12 schrieb KG5IEN:

**Comment by KG5IEN** [comment]
>Hmm... but as I wrote (assumed), the intention of the parameter "log" is
*NOT* being a char pointer, but a single char.
So this fix would be wrong. Again, I *guess* what Mike meant was a
single char, i.e.:
>
>void print_time_hook( char log ).
Note the lack of the asterisk.

I understand this, I was just calling it out explicitly in my comment.  I'd prefer that issue #732 be discussed and resolved there rather that in this PR.  I can remove my changes to this file if that would allow the rest of the improvements to move forward while further investigation of 8f8ca60 continues in parallel.

**Comment by KG5IEN** [comment]
Nevermind, I decided to resolve #732 as well as the other warnings introduced in 8f8ca60 in this PR.  See a0e2037.

**Comment by travisgoodspeed** [comment]
@KG5IEN Is this ready to review and merge?

@Wolf-DL4YHF Would merging this PR delay merging your Morse PR?  Do you see anything wrong with this patch?

If not, I'll try to merge this PR tomorrow.

**Comment by KG5IEN** [comment]
@travisgoodspeed Yes, ready to review and merge.

**Comment by Wolf-DL4YHF** [comment]
Hi Travis and all,

I don't think merging this PR will delay anything. Except for the 
"const" specified (which I will add in my own branch now to avoid future 
trouble), the prototype / declaration of print_time_hook would be the 
same as PR #726 .

What I am still  (a bit) concerned about is this:
Since "print_time_hook" looks like one of the special functions designed 
to replace something in the original firmware,
will THE ORIGINAL FIRMWARE call it from somewhere ? If it did, it would 
hardly pass in a 'char' as a function argument in register R0, and the 
content of R0 (= char log) would be unpredictable then.
In other words: If print_time_hook replaces a function of the original 
firmware, we should not change the function's argument list at all.
Well, I didn't have the time to look it up in the symbol file or in the 
disassembled original firmware yet, to examine the call stack for 
print_time_hook() .


Cheers,
   Wolf .

Am 19.04.2017 um 05:09 schrieb Travis Goodspeed:

**Comment by KG5IEN** [comment]
> @Wolf-DL4YHF wrote:
> Well, I didn't have the time to look it up in the symbol file

Looking through the symbols files (both the ones in applet/src and those created by symgrate in the symbols subdir), I see no references to print_time_hook.

**Comment by KG5IEN** [comment]
Hi @travisgoodspeed ,
Just checking in: I still believe this is ready to review and merge.  If any changes are desired or necessary, just let me know and I'll get to work addressing them.

**Comment by travisgoodspeed** [comment]
Thank you kindly!
